compactor/src/compact.rs

```diff
@@ -460,20 +467,39 @@ impl Compactor {
         Ok(())
     }
 
-    // Group given parquet files into partition of the same (sequencer_id, table_id, partition_id)
-    fn group_parquet_files_into_partition(
-        parquet_files: Vec<ParquetFile>,
-    ) -> BTreeMap<TablePartition, Vec<ParquetFile>> {
-        let mut groups: BTreeMap<TablePartition, Vec<ParquetFile>> = BTreeMap::default();
-        for file in parquet_files {
-            let key = TablePartition::new(file.sequencer_id, file.table_id, file.partition_id);
-            if let Some(val) = groups.get_mut(&key) {
-                val.push(file);
+    // Group time-contiguous non-overlapped groups if their total size is smaller than a threshold
+    fn group_small_contiguous_groups(
+        file_groups: &mut Vec<GroupWithMinTimeAndSize>,
+    ) -> Vec<Vec<ParquetFile>> {
```

Comment:
Because `file_groups` isn't used after this function is called, the return value is used, this function can take ownership of `file_groups` to better communicate what this function is doing:

```suggestion
    fn group_small_contiguous_groups(
        mut file_groups: Vec<GroupWithMinTimeAndSize>,
    ) -> Vec<Vec<ParquetFile>> {
```

Then `let mut overlapped_file_groups`  above and everywhere else the vecs are initialized don't need to be mutable, and everywhere that passes `&mut` to this function can just pass ownership.